### DISCUSSION TITLE
Build fails because of Swift

### DISCUSSION THREAD

**Discussion by code-on-git** [actions/virtual-environments#9963] [General questions] [answered]
I've seen both an [issue](https://github.com/actions/runner-images/issues/9664) and [discussion](https://github.com/actions/runner-images/discussions/9802) about this error and it seems to have been closed but I still get the same error even when I clone directly from https://github.com/actions/runner-images.git
```
==> azure-arm.build_image: Downloading package from https://swift.org/builds/swift-5.10-release/ubuntu2204/swift-5.10-RELEASE/swift-5.10-RELEASE-ubuntu22.04.tar.gz to /tmp/swift-5.10-RELEASE-ubuntu22.04.tar.gz...
==> azure-arm.build_image: Package downloaded in 8 seconds
==> azure-arm.build_image: gpg: directory '/root/.gnupg' created
==> azure-arm.build_image: gpg: keybox '/root/.gnupg/pubring.kbx' created
==> azure-arm.build_image: gpg: keyserver receive failed: Connection timed out
==> azure-arm.build_image: Script exited with non-zero exit status: 2. Allowed exit codes are: [0]
```

Is it something that I need to do to fix this or is it still an issue?

**Comment by lukasbbuehler** [comment]
Same for me still an issue!

**Reply by lukasbbuehler** [reply, marked as answer]
We noticed a change in the swift script: Download ist now from gpg --keyserver hkp://keyserver.ubuntu.com \ instead of pgp_key_path=$(download_with_retry "https://swift.org/keys/all-keys.asc")
gpg --no-default-keyring --keyring swift --import "$pgp_key_path"

The issue is that hkp:// is using the default port 11371, so for us the firewall is blocking this. As a solution you could change it to gpg --keyserver hkp://keyserver.ubuntu.com:80 so the request will go over port 80 which is usually allowed.

**Reply by lukasbbuehler** [reply]
Atleast for us this worked. But it would be great to have this changed.

**Reply by lukasbbuehler** [reply]
I created a PR for it: https://github.com/actions/runner-images/pull/10007

**Reply by code-on-git** [reply]
Great finding! Works perfectly

**Reply by lukasbbuehler** [reply]
@code-on-git  Did a bit more testing, to improve security. It will also work if you use hkps over port 443 (hkps://keyserver.ubuntu.com:443) so you use encryption.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
